pkg/actors/actors.go

```diff
@@ -1081,26 +1455,31 @@ func (a *actorsRuntime) DeleteReminder(ctx context.Context, req *DeleteReminderR
 	}
 
 	err := backoff.Retry(func() error {
-		reminders, remindersEtag, err := a.getRemindersForActorType(req.ActorType)
+		reminders, actorMetadata, err := a.getRemindersForActorType(req.ActorType, true)
 		if err != nil {
 			return err
 		}
 
+		// remove from partition first.
+		remindersInPartition, stateKey, etag := actorMetadata.removeReminderFromPartition(reminders, req.ActorType, req.ActorID, req.Name)
+
+		// now, we can remove from the "global" list.
 		for i := len(reminders) - 1; i >= 0; i-- {
-			if reminders[i].ActorType == req.ActorType && reminders[i].ActorID == req.ActorID && reminders[i].Name == req.Name {
+			if reminders[i].reminder.ActorType == req.ActorType && reminders[i].reminder.ActorID == req.ActorID && reminders[i].reminder.Name == req.Name {
 				reminders = append(reminders[:i], reminders[i+1:]...)
```

Comment:
Should use a lock here?

Reply:
reminders is local variable while a.reminders (used later on) is the shared field with the lock. So, I do not see the need to lock the local variable here. Please, let me know if I am missing something.

Reply:
ok, I thought it was a reference to `a.reminders`, I see it's not.